DuckDuckGo/MainViewController.swift

```diff
@@ -225,7 +225,11 @@ class MainViewController: UIViewController {
             omniBar.stopBrowsing()
             return
         }
-        omniBar.refreshText(forUrl: tab.link?.url)
+
+        if !tab.isError {
+            omniBar.refreshText(forUrl: tab.link?.url)
```

Comment:
Could this be encapsulated in the tab itself so that when we ask the tab to provide the correct url it provides that (perhaps via its model) irrespective of error state?

There is generally some unexpected error behavior, for example:

If a tab has loaded a page successfully and subsequently goes to an error then:
* Selecting `refresh` goes to previous successful page (rather than trying to reload the page which errored)
* If the app is closed on an error page, it relaunches to the previous page

If a fresh tab is opened and the user tries to open a url which errors:
* Pressing `Refresh` does nothing

Of course you may find another way to solve these.